File: ProEvento/ProEventos.Domain/Models/Palestrante.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ProEventos.Domain.Models
{
    public class Palestrante
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string Descricao { get; set; }
        public string ImagemURL { get; set; }
        public string Telefone { get; set; }
        public string Email { get; set; }
        public IEnumerable<RedeSocial> RedeSocials { get; set; }
        public IEnumerable<PalestranteEvento> PalestrantesEventos { get; set; }
    }
}

File: ProEvento/ProEventos/src/app/models/Palestrante.ts
import { UserUpdate } from "./Identity/UserUpdate";
import { RedeSocial } from "./RedeSocial"

export interface Palestrante {
  id: number;
  miniCurriculo: string;
  user: UserUpdate;
  redeSocials: RedeSocial[];
  palestrantesEventos: Palestrante[];
}
